src/app/bundles/document-selection/document-selection.component.ts

```diff
@@ -8,11 +9,38 @@ import { MockCaseDocuments } from '../shared/mockcasedocumentdata';
 })
 export class DocumentSelectionComponent implements OnInit {
 
-  cases = MockCaseDocuments;
+  documents = MockCaseDocuments;
+  selectAllStatus: boolean;
+
+  @ViewChildren('documentItem') documentItem: QueryList<DocumentItemComponent>;
 
   constructor() { }
 
   ngOnInit() {
+    this.selectAllStatus = false;
+  }
+
+  updateSelectAll() {
+    this.selectAllStatus = !this.selectAllStatus;
+    this.documentItem.map((document: DocumentItemComponent) => document.checked = this.selectAllStatus);
   }
 
+  isAllCheckboxSelected() {
+    this.selectAllStatus = true;
+    this.documentItem.map((document: DocumentItemComponent) => {
+      if (document.checked === false) {
+        this.selectAllStatus = false;
+      }
+    });
+  }
+
+  selectedCheckbox() {
+    let arr1 = [];
+    this.documentItem.map((document: DocumentItemComponent) => {
```

Comment:
this could be implemented as a filter... `return this.documentItem.filter(document => document.checked);`

Reply:
but also... what is this doing? Is it debug code?

Reply:
as there is no back end, for now this is printing the selected documents.